cloudinit/config/schemas/cc_apt_configure.yaml

```diff
@@ -0,0 +1,268 @@
+            present in ``sources.list`` it will be ignored. For convenience, several aliases are provided for
+            ``disable_suites``:
+
+              - ``updates`` => ``$RELEASE-updates``
+              - ``backports`` => ``$RELEASE-backports``
+              - ``security`` => ``$RELEASE-security``
+              - ``proposed`` => ``$RELEASE-proposed``
+              - ``release`` => ``$RELEASE``.
+
+            When a suite is disabled using ``disable_suites``, its entry in ``sources.list`` is not deleted; it
+            is just commented out.
+        primary:
+          description: |
+            The primary and security archive mirrors can be specified using the ``primary`` and
+            ``security`` keys, respectively. Both the ``primary`` and ``security`` keys take a list
+            of configs, allowing mirrors to be specified on a per-architecture basis. Each config is a
+            dictionary which must have an entry for ``arches``, specifying which architectures
+            that config entry is for. The keyword ``default`` applies to any architecture not
+            explicitly listed. The mirror url can be specified with the ``uri`` key, or a list of mirrors to
+            check can be provided in order, with the first mirror that can be resolved being selected. This
+            allows the same configuration to be used in different environment, with different hosts used
+            for a local apt mirror. If no mirror is provided by ``uri`` or ``search``, ``search_dns`` may be
+            used to search for dns names in the format ``<distro>-mirror`` in each of the following:
+
+                - fqdn of this host per cloud metadata,
+                - localdomain,
+                - domains listed in ``/etc/resolv.conf``.
+
+            If there is a dns entry for ``<distro>-mirror``, then it is assumed that there is a distro mirror
+            at ``http://<distro>-mirror.<domain>/<distro>``. If the ``primary`` key is defined, but not the
+            ``security`` key, then then configuration for ``primary`` is also used for ``security``.
+            If ``search_dns`` is used for the ``security`` key, the search pattern will be
+            ``<distro>-security-mirror``.
+
+            Each mirror may also specify a key to import via any of the following optional keys:
+
+                - ``keyid``: a key to import via shortid or fingerprint.
+                - ``key``: a raw PGP key.
+                - ``keyserver``: alternate keyserver to pull ``keyid`` key from.
+
+            If no mirrors are specified, or all lookups fail, then default mirrors defined in the datasource
+            are used. If none are present in the datasource either the following defaults are used:
+
+                - ``primary`` => ``http://archive.ubuntu.com/ubuntu``.
+                - ``security`` => ``http://security.ubuntu.com/ubuntu``
+          type: "array"
+          items: &MIRROR_PROPERTY
```

Comment:
This will probably be the route to go for reuse. I tried the defs/ref route, but it wouldn't work for me.

Reply:
I was able to get defs/refs working with a single JSON file: https://github.com/TheRealFalcon/cloud-init/pull/3/files

The one thing I wonder about with both YAML and JSON is the loss of common definitions between python on the YAML/JSON  for defaults etc.  It just means we either have to source the definitions from the source YAML/JSON in python, or just make sure the two stay in sync.